daprdocs/content/en/reference/components-reference/supported-state-stores/setup-azure-cosmosdb.md

```diff
@@ -127,10 +131,102 @@ curl -X POST http://localhost:3500/v1.0/state/<store_name> \
+
+## Setting up Cosmos DB for authenticating with Azure AD
+
+When using the Dapr Cosmos DB state store and authenticating with Azure AD, you need to perform a few additional steps to set up your environment.
+
+Prerequisites:
+
+- You will need a Service Principal created as per the instructions in the [authenticating to Azure]({{< ref authenticating-azure.md >}}) page. You will need the ID of the Service Principal for the commands below (note that this is different from the client ID of your application, or the value you use for `azureClientId` in the metadata).
+- [Azure CLI](https://docs.microsoft.com/cli/azure/install-azure-cli)
+- [jq](https://stedolan.github.io/jq/download/)
+- The scripts below are optimized for a bash or zsh shell
+
+### Granting your Azure AD application access to Cosmos DB
+
+> You can find more information on the [official documentation](https://docs.microsoft.com/en-us/azure/cosmos-db/how-to-setup-rbac), including instructions to assign more granular permissions.
+
+In order to grant your application permissions to access data stored in Cosmos DB, you need to assign it a custom role for the Cosmos DB data plane. In this example you're going to use a built-in role, "Cosmos DB Built-in Data Contributor", which grants your application full read-write access to the data; you can optionally create custom, fine-tuned roles following the instructions in the official docs.
+
+```sh
+# Name of the Resource Group that contains your Cosmos DB
+RESOURCE_GROUP="..."
+# Name of your Cosmos DB account
+ACCOUNT_NAME="..."
+# ID of your Service Principal object
+PRINCIPAL_ID="..."
+# ID of the "Cosmos DB Built-in Data Contributor" role
+# You can also use the ID of a custom role
+ROLE_ID="00000000-0000-0000-0000-000000000002"
+
+az cosmosdb sql role assignment create \
+  --account-name "$ACCOUNT_NAME" \
+  --resource-group "$RESOURCE_GROUP" \
+  --scope "/" \
+  --principal-id "$PRINCIPAL_ID" \
+  --role-definition-id "$ROLE_ID"
+```
+
+### Creating the stored procedures for Dapr
+
+When using Cosmos DB as a state store for Dapr, we need to create two stored procedures in your collection. When you configure the state store using a "master key", Dapr creates those for you, automatically. However, when your state store authenticates with Cosmos DB using Azure AD, because of limitations in the platform we are not able to do it automatically.
+
+If you are using Azure AD to authenticate your Cosmos DB state store and have not created the stored procedures (or if you are using an outdated version of them), your Dapr sidecar will fail to start and you will see an error similar to this in your logs:
+
+```text
```

Comment:
I am including here the error message in full for SEO. So when people see this error and Google it, they can land on this page.